test/server/cards/01_SOR/units/BlackOneScourgeOfStarkillerBase.spec.ts

```diff
@@ -0,0 +1,67 @@
+describe('Black One', function() {
+    integration(function(contextRef) {
+        describe('ISB Agent\'s When Played and When Defeated ability', function() {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['black-one#scourge-of-starkiller-base'],
+                        deck: ['confiscate', 'waylay', 'isb-agent', 'cartel-spacer', 'wampa', 'disarm', 'atst']
+                    },
+                    player2: {
+                        hand: ['vanquish', 'rivals-fall']
+                    }
+                });
+            });
+
+            it('draw 3 cards if hand is discarded', function () {
+                const { context } = contextRef;
+
+                // Player 1 plays Black One and activates When Played, drawing cards
+                context.player1.clickCard(context.blackOne);
+
+                expect(context.player1.handSize).toBe(0);
+                expect(context.player1).toHaveEnabledPromptButton('You may discard your hand');
```

Comment:
Can we use `toHavePassAbilityPrompt` which will confirm the name and that there's a pass button